ai-catalog/samples/gemini-live-todo/src/main/java/com/android/ai/samples/geminilivetodo/ui/TodoScreen.kt

```diff
@@ -0,0 +1,225 @@
+                        targetValue = MaterialTheme.colorScheme.primary.copy(alpha = 0.9f),
+                        animationSpec = infiniteRepeatable(
+                            animation = tween(1000, easing = LinearEasing),
+                            repeatMode = RepeatMode.Reverse,
+                        ),
+                        label = "mic_color",
+                    ).value
+                } else {
+                    MaterialTheme.colorScheme.primaryContainer
+                }
+
+                FloatingActionButton(
+                    onClick = { if (successState.isLiveSessionReady) viewModel.toggleLiveSession() },
+                    containerColor = containerColor,
+                ) {
+                    Icon(micIcon, stringResource(R.string.interact_with_todolist_by_voice))
+                }
+            }
+        },
+        floatingActionButtonPosition = FabPosition.Center,
+        modifier = Modifier.fillMaxSize(),
+    ) { paddingValues ->
+        Column(
+            modifier = Modifier
+                .padding(paddingValues)
+                .padding(16.dp)
+                .imePadding()
+                .fillMaxSize(),
+        ) {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(bottom = 16.dp),
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                OutlinedTextField(
+                    value = text,
+                    onValueChange = { text = it },
+                    label = { Text(stringResource(R.string.new_task_placeholder)) },
+                    modifier = Modifier.weight(1f),
+                    singleLine = true,
+                )
+                Spacer(modifier = Modifier.width(8.dp))
+                Button(
+                    onClick = {
+                        if (text.isNotBlank()) {
+                            viewModel.addTodo(text)
+                            text = ""
+                        }
+                    },
+                ) {
+                    Text(stringResource(R.string.add_button))
+                }
+            }
+
+            when (uiState) {
+                is TodoScreenUiState.Initial -> {
+                    // Show a loading indicator or initial state
+                }
+                is TodoScreenUiState.Success -> {
+                    val todos = (uiState as TodoScreenUiState.Success).todos
+                    LazyColumn(modifier = Modifier.fillMaxSize()) {
+                        items(todos.reversed(), key = { it.id }) { todo ->
+                            TodoItem(
+                                task = todo,
+                                onToggle = { viewModel.toggleTodoStatus(todo.id) },
+                                onDelete = { viewModel.removeTodo(todo.id) },
+                            )
+                            HorizontalDivider()
```

Comment:
nit: It's pretty common to use itemsIndexed to have the index and then not add a divider after the last item